code/modules/xenoarcheaology/effect_master.dm

```diff
@@ -0,0 +1,383 @@
+
+		if (istype(W, /obj/item/weapon/reagent_containers))
+			if(W.reagents.has_reagent("hydrogen", 1) || W.reagents.has_reagent("water", 1))
+				if(my_effect.trigger == TRIGGER_WATER)
+					my_effect.ToggleActivate()
+			else if(W.reagents.has_reagent("sacid", 1) || W.reagents.has_reagent("pacid", 1) || W.reagents.has_reagent("diethylamine", 1))
+				if(my_effect.trigger == TRIGGER_ACID)
+					my_effect.ToggleActivate()
+			else if(W.reagents.has_reagent("phoron", 1) || W.reagents.has_reagent("thermite", 1))
+				if(my_effect.trigger == TRIGGER_VOLATILE)
+					my_effect.ToggleActivate()
+			else if(W.reagents.has_reagent("toxin", 1) || W.reagents.has_reagent("cyanide", 1) || W.reagents.has_reagent("amatoxin", 1) || W.reagents.has_reagent("neurotoxin", 1))
+				if(my_effect.trigger == TRIGGER_TOXIN)
+					my_effect.ToggleActivate()
+		else if(istype(W,/obj/item/weapon/melee/baton) && W:status ||\
+				istype(W,/obj/item/weapon/melee/energy) ||\
+				istype(W,/obj/item/weapon/melee/cultblade) ||\
+				istype(W,/obj/item/weapon/card/emag) ||\
+				istype(W,/obj/item/device/multitool))
+			if (my_effect.trigger == TRIGGER_ENERGY)
+				my_effect.ToggleActivate()
+
+		else if (istype(W,/obj/item/weapon/flame) && W:lit ||\
+				istype(W,/obj/item/weapon/weldingtool) && W:welding)
+			if(my_effect.trigger == TRIGGER_HEAT)
+				my_effect.ToggleActivate()
+		else
+			..()
+			if (my_effect.trigger == TRIGGER_FORCE && W.force >= 10)
+				my_effect.ToggleActivate()
+
+/datum/component/artifact_master/proc/on_moved()
+	for(var/datum/artifact_effect/my_effect in my_effects)
+		if(my_effect)
+			my_effect.UpdateMove()
+
+/datum/component/artifact_master/process()
+	var/turf/L = holder.loc
+	if(!istype(L) && !isliving(L)) 	// We're inside a non-mob container or on null turf, either way stop processing effects
+		return
+
+	if(istype(holder, /atom/movable))
+		var/atom/movable/HA = holder
+		if(HA.pulledby)
+			on_bumped(HA.pulledby)
+
+	//if any of our effects rely on environmental factors, work that out
+	var/trigger_cold = 0
+	var/trigger_hot = 0
+	var/trigger_phoron = 0
+	var/trigger_oxy = 0
+	var/trigger_co2 = 0
+	var/trigger_nitro = 0
+
+	var/turf/T = get_turf(holder)
+	var/datum/gas_mixture/env = T.return_air()
+	if(env)
+		if(env.temperature < 225)
+			trigger_cold = 1
+		else if(env.temperature > 375)
+			trigger_hot = 1
+
+		if(env.gas["phoron"] >= 10)
+			trigger_phoron = 1
+		if(env.gas["oxygen"] >= 10)
+			trigger_oxy = 1
+		if(env.gas["carbon_dioxide"] >= 10)
+			trigger_co2 = 1
+		if(env.gas["nitrogen"] >= 10)
+			trigger_nitro = 1
+
+	for(var/datum/artifact_effect/my_effect in my_effects)
+		my_effect.artifact_id = artifact_id
+
+		my_effect.process()
+
+		//COLD ACTIVATION
+		if(trigger_cold)
+			if(my_effect.trigger == TRIGGER_COLD && !my_effect.activated)
+				my_effect.ToggleActivate()
+		else
+			if(my_effect.trigger == TRIGGER_COLD && my_effect.activated)
+				my_effect.ToggleActivate()
```

Comment:
All of these checks specifically want to perform the activation when one condition is true and the other is false, so you could just use XOR
```suggestion
if(my_effect.trigger == TRIGGER_COLD && (trigger_cold ^ my_effect.activated))
	my_effect.ToggleActivate()
```
That is, it'll run on `(A & !B) | (!A & B)` ~= `!((A & B) | (!A & !B))`